src/Test.EndToEnd.Baselining/Test.EndToEnd.Baselining.csproj

```diff
@@ -14,5 +14,6 @@
     <OutputType>Exe</OutputType>
     <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
     <IsPackable>false</IsPackable>
+    <RootNamespace>Test.EndToEnd.Baselining</RootNamespace>
```

Comment:
>RootNamespace [](http://example.com/codeflow?start=5&length=13)

The root namespace is used as a common prefix by the resource compiler. And so we create a root namespace that matches the binary name. This makes it very easy at runtime to locate embedded resources, etc., we simply look at the Assembly binary name to do that.

There is an unpleasant consequence of this 'trick', whenever you create a new file, this will be used as the default namespace (and it never should be retained). So that's an inconvenience for sure.

VS has some knobs for controlling the resource manifest name, e.g., 'Custom Tool Namespace' in the properties pane. This approach can be tedious to use and VS wants to get involved. I haven't found a clean approach. If someone else knows more than I, let me know. :)

Reply:
I think, the issue is related to this:
[https://github.com/microsoft/sarif-sdk/blob/ed38f391b85e9a26b5e6079ad29fdd9f2b342e5e/src/build.props#L40-L41](https://github.com/microsoft/sarif-sdk/blob/ed38f391b85e9a26b5e6079ad29fdd9f2b342e5e/src/build.props#L40-L41)

And, each test project we are also using the same build.props, which is changing the namespace.

We could create a condition in that build props to only apply that if we have a product code or create a separate props file to test projects.

Reply:
Thanks, Eddy, something for the future.